homeassistant/components/fatek/light.py

```diff
@@ -0,0 +1,116 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import FatekEntity, FatekServer, FatekStateCoordinator
+from .const import DOMAIN, FATEK_INDEX, FATEK_REGISTER
+
+LOGGER = logging.getLogger("Fatek")
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(FATEK_REGISTER, default="M"): str,
+        vol.Required(FATEK_INDEX, default=0): int,
+        vol.Required(CONF_NAME, default=""): str,
+    }
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Perform the setup for FatekLight devices."""
+    name = config[CONF_NAME]
+    register = config[FATEK_REGISTER]
+    index = config[FATEK_INDEX]
+    entry_id: str = hass.data[DOMAIN]["entry_id"]
+    if len(entry_id) == 32:
+        coordinator: FatekStateCoordinator = hass.data[DOMAIN][entry_id]
+        add_entities([FatekLight(coordinator, name, register, index)], True)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Utworzenie encji."""
+    if hass.data[DOMAIN]["coordinator_running"] is False:
+        hass.data[DOMAIN]["coordinator_running"] = True
+        entry_id = config.entry_id
+        coordinator: FatekStateCoordinator = hass.data[DOMAIN][entry_id]
+        async_add_entities([FatekServer(coordinator, entry_id)])
+
+
+class FatekLight(LightEntity, FatekEntity):
+    """Światło sterowane przez Fateka."""
+
+    def __init__(
+        self,
+        coordinator: FatekStateCoordinator,
+        name: str,
+        register: str,
+        index: int,
+    ) -> None:
+        """Initialize the FatekLight."""
+        LOGGER.info("Initializing Fatek.Light: %s_%s ", register, index)
+        #        super().__init__()
+        LightEntity.__init__(self)
+        FatekEntity.__init__(self, coordinator, name, register, index)
+        self._state: bool = False
+
+    def set_entity_state(self, new_state) -> bool:
+        """Ustawienie stanu."""
+        # LOGGER.info("FatekLight: set_state: %s", self.entity_id)
+        if self.hass.states.get(self.entity_id):
+            if new_state == 1:
+                self.hass.states.async_set(self.entity_id, "on")
+            else:
+                self.hass.states.async_set(self.entity_id, "off")
+        return True
+
+    def update(self) -> None:
+        """Synchroniuzj stan."""
+        # LOGGER.info("FatekLight: update: %s ", self.entity_id)
+        b_new_state = self._coordinator.get_m_register(self._index)
+        # LOGGER.info(b_new_state)
+        if b_new_state != self._state:
```

Comment:
Is this `if` necessary – wouldn't it be safer to just set the state always based on the known state from the coordinator?